src/test/java/io/managed/services/test/kafka/KafkaMgmtAPITest.java

```diff
@@ -218,6 +218,123 @@ public void testCreateTopics() {
         kafkaInstanceApi.createTopic(metricTopicPayload);
     }
 
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        log.info("Max partition limit per given instance: {}", maxPartitionLimit);
+
+        log.info("Attempt to create topic '{}', with too many partitions: {}", topicName, maxPartitionLimit + 1);
+        var payload = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit + 1));
+        assertThrows(ApiGenericException.class,
+                () -> kafkaInstanceApi.createTopic(payload));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedParitionLimitByTopicConfiguration() {
+
+        var topicName = "topic-part-exceed-conf";
+        KafkaInstanceApiUtils.applyTopic(kafkaInstanceApi, topicName);
+
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        assertThrows(ApiGenericException.class,
+                () -> KafkaInstanceApiUtils.updateTopicPartition(kafkaInstanceApi, topicName, maxPartitionLimit + 1));
```

Comment:
I don't think this is going to throw the exception. In the CustomAclAuthorizer, we don't have the ability to inspect the request to see if it would result in a limit breach. We can only deny requests for new partitions if the limit has _already_ been reached (which I don't think it has been at this point).

WDYT?

Reply:
My understanding is that there are two way of making sure we will not breach the Limit:
1) extension you provided (sums of all partitions)
2) limiting every single given Topic from breaching counts directly e.g., I have global partition limit set to 100, no matter how many partitions i have already created, I will still not be able to create topic with 101 partitions on its own.

Reply:
The two extensions we use are:
- CreateTopicPolicy: if a new topic request either exceeds the limit by itself, or if adding it to the current partition count would exceed the limit, then the request will be rejected.
- Authorizer: if the current partition count is either at or above the limit, then requests to increase partitions will be rejected.

It's only in the requests that go through the CreateTopicPolicy that we can inspect how many partitions are being requested. On the Authorizer path (increase partitions on existing topic), we can't — we can only deny requests when the limit has already been hit. So that's why I think this one will not result in a denial/exception.

We're aware it's imperfect, but it's the best we can do with the current controls available to us.

Reply:
but isnt the test matching the first point you just wrote, i.e., newly created Topic requires 1501 partition whereas the limit is 1500  ?  "adding it to the current partition count would exceed the limit, then the request will be rejected."

Reply:
I don't believe so. My understanding is that L242 creates a topic with 1 partition, then this L246 does an _update_ of that topic to increase the number of partitions to 1501, right?

I think the case you're thinking of where a new topic request is trying to use 1501 partitions when the limit is 1500 is the `testForbiddenToExceedPartitionLimitByTopicCreation` test above it, rather than this one.